src/control/settings/Settings.cpp

```diff
@@ -2005,20 +2005,17 @@ void Settings::setSidebarVisible(bool visible)
 	save();
 }
 
-int Settings::getSidebarWidth()
+int Settings::getSidebarWidth() const
 {
 	XOJ_CHECK_TYPE(Settings);
 
-	if (this->sidebarWidth < 50)
-	{
-		setSidebarWidth(150);
-	}
-	return this->sidebarWidth;
+	return this->sidebarWidth < 50 ? 150 : this->sidebarWidth;
```

Comment:
Is the behaviour of this method changed by accident or was this done with purpose? It saves against buggy settings files which may be out there.

Reply:
It was indeed done by purpose. The check should happen on write, not on read. I changed both functions.

Reply:
The thing is that if there is a buggy settings file out there then it will create an invalid state for the Xournal++ instance even if writing the file back in a correct way. You should definitely check this when reading the settings file. But it might be better to place this in the constructor than in the `getSidebarWidth()` method as it is a one-time fix.

Reply:
Or I have to be Shure, that this field is only written by set...(value).

Reply:
We are talking of files written with prior versions of Xournal++. Backwards-compatibility :wink:

Reply:
Year I know what you mean, I moved the check to the setter, but if the setter isn't used to initialize the value on the load of old files, I need to make Shure it does, or use the check there also.

Reply:
I am back home in 10 minutes, then I can check everything